src/fiber.cr

```diff
@@ -223,6 +228,16 @@ class Fiber
     Crystal::Scheduler.enqueue(self)
   end
 
+  {% if flag?(:linux) || flag?(:solaris) %}
+    # :nodoc:
+    def timerfd : Crystal::System::TimerFD
+      # we keep a single timerfd per fiber and use it for all event types (:sleep,
```

Comment:
uhh. This can create a LOT of file descriptors. Perhaps not a problem, but it definitely could be. I woe the person implementing sleepsort using this :sweat_smile: 

On the other hand it keeps them separate from each other - otherwise you'd have to have something like the timeout solution on windows with having one timer and a sorted list (or well, a heap would be pretty ideal for this, but we are not there yet). Ew.

Reply:
That's only one timerfd per fiber, lazily allocated, so it shouldn't be _that_ many fibers. It's still better than creating/closing a timerfd on each sleep/timeout :cold_sweat: 

Eventually, yes, I'd like to extract them into a `Timers` priority queue, and have the schedulers dequeue them when ready, so we'd only have 1 timerfd per epoll instance for sub-millisecond precision timeouts.

Reply:
There are now only 1 timer per eventloop.

I'm not exactly sure how it will behave with the MT execution context of [RFC #2](https://github.com/crystal-lang/rfcs/pull/2), where I'd like to have one EL per EC, but I'll see that next week.